src/hint_processor/builtin_hint_processor/secp/bigint_utils.zig

```diff
@@ -0,0 +1,110 @@
+const std = @import("std");
+const Felt252 = @import("../../../math/fields/starknet.zig").Felt252;
+const Relocatable = @import("../../../vm/memory/relocatable.zig").Relocatable;
+const CairoVM = @import("../../../vm/core.zig").CairoVM;
+const HintError = @import("../../../vm/error.zig").HintError;
+const hint_utils = @import("../../hint_utils.zig");
+const HintReference = @import("../../hint_processor_def.zig").HintReference;
+const ApTracking = @import("../../../vm/types/programjson.zig").ApTracking;
+const ExecutionScopes = @import("../../../vm/types/execution_scopes.zig").ExecutionScopes;
+const pow2ConstNz = @import("../../math_hints.zig").pow2ConstNz;
+const packBigInt = @import("../../uint_utils.zig").pack;
+const splitBigInt = @import("../../uint_utils.zig").split;
+
+pub const BigInt3 = BigIntN(3);
+pub const Uint384 = BigIntN(3);
+pub const Uint512 = BigIntN(4);
+pub const BigInt5 = BigIntN(5);
+pub const Uint768 = BigIntN(6);
+
+pub fn BigIntN(comptime NUM_LIMBS: usize) type {
+    return struct {
+        limbs: [NUM_LIMBS]Felt252 = undefined,
+        pub fn fromBaseAddr(self: BigIntN(NUM_LIMBS), addr: Relocatable, vm: *CairoVM) !BigIntN(NUM_LIMBS) {
+            for (0..NUM_LIMBS) |i| {
+                self.limbs[i] = vm.getFelt(addr + i) catch return HintError.IdentifierHasNoMember;
+            }
+            return BigIntN{ .limbs = self.limbs };
+        }
+
+        pub fn fromVarName(self: BigIntN(NUM_LIMBS), name: []const u8, vm: *CairoVM, ids_data: std.StringHashMap(HintReference), ap_tracking: ApTracking) !BigIntN(NUM_LIMBS) {
+            const baseAddress = try hint_utils.getRelocatableFromVarName(name, vm, ids_data, ap_tracking);
+            return self.fromBaseAddr(baseAddress, vm);
+        }
+
+        pub fn fromValues(limbs: [NUM_LIMBS]Felt252) !BigIntN(NUM_LIMBS) {
+            return BigIntN{ .limbs = limbs };
+        }
+
+        pub fn insertFromVarName(allocator: std.mem.allocator, var_name: []const u8, vm: *CairoVM, ids_data: std.StringHashMap(HintReference), ap_tracking: ApTracking) !void {
+            const addr = try hint_utils.getRelocatableFromVarName(var_name, vm, ids_data, ap_tracking);
+            for (0..NUM_LIMBS) |i| {
+                try vm.insertInMemory(allocator, addr.offset(i), .limbs[i]);
```

Comment:
- I don't think that `addr.offset(i)` is correct because `hint_utils.getRelocatableFromVarName` will return you a relocatable address with a single offset no?
- I think that you should add `self` as an argument of the function in order to have access to `self.limbs[i]` or did I miss something?